jr-test-module/src/test/java/Java14RecordTest.java

```diff
@@ -0,0 +1,22 @@
+import com.fasterxml.jackson.jr.ob.JSON;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * This test is in test module since the JDK version to be tested is higher than other, and hence supports Records.
+ */
+public class Java14RecordTest {
```

Comment:
Records only officially part of 15 I think (in 14 preview feature) so let's call Java17 (first Long-Term Supported (LTS) JDK that has Records).